src/persist/src/indexed/cache.rs

```diff
@@ -293,14 +294,30 @@ impl<B: Blob> BlobCache<B> {
 
     /// Fetches metadata about what batches are in [Blob] storage.
     pub fn get_meta(&self) -> Result<Option<BlobMeta>, Error> {
-        let blob = self.blob.lock()?;
+        let mut blob = self.blob.lock()?;
         let bytes = match block_on(blob.get(Self::META_KEY))? {
             Some(bytes) => bytes,
             None => return Ok(None),
         };
-        let meta = BlobMeta::decode(&bytes).map_err(|err| {
-            Error::from(format!("invalid meta at key {}: {}", Self::META_KEY, err))
-        })?;
+
+        let meta = match BlobMeta::decode(&bytes) {
+            Ok(meta) => meta,
+            Err(CodecError::InvalidEncodingVersion(_)) => {
```

Comment:
This isn't a great spot for this check. I think the contract is that it should happen at most once at startup (and probably only in the production startup codepath). In theory, this should only get hit once as written (b/c it'll get written back out with the current version), but for something as scary as deleting all a user's persisted data, it's not as dead simple to reason about as I'd like.

Concretely, I think it should live in `coord::persistcfg::PersistConfig::Init`

Reply:
Agree that this is not a good spot for the check and relies way too much on the incidental fact that this function currently only gets called during indexed initialization. That was an oversight on my part.

Let me noodle on where to put this check -- I sort of think maybe `runtime::start` is a good place for it (I feel weird putting it totally outside of the indexed / runtime code because that feels a little abstraction breaking-y)

Reply:
On the contrary, I picked `coord::persistcfg::PersistConfig::Init` precisely because this hack itself is abstraction breaking (and temporary). IMO src/persist isn't the right place for it